### PR TITLE
refactor(emp clients): Optimize parallel promise calls

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

The EMP clients do many on-chain calls that can be run in parallel. This PR applies the same refinements as done to other clients to the event client.

**Summary**

Move calls into `promise.All`.

### CODE REVIEW

File: financial-templates-lib/clients/ExpiringMultiPartyEventClient.js [hunk 1]
```diff
@@ -113,17 +113,50 @@ class ExpiringMultiPartyEventClient {
...
-    const lastBlockToSearch = this.lastBlockToSearchUntil
-      ? this.lastBlockToSearchUntil
-      : await this.web3.eth.getBlockNumber();
-
-    // Look for events on chain from the previous seen block number to the current block number.
-    // Liquidation events
-    const liquidationEventsObj = await this.emp.getPastEvents("LiquidationCreated", {
+    const [currentBlockNumber, currentTime] = await Promise.all([
+      this.web3.eth.getBlockNumber(),
+      this.emp.methods.getCurrentTime().call()
```
Comment (nicholaspai) [inline financial-templates-lib/clients/ExpiringMultiPartyEventClient.js:118]: I think we should move the `getCurrentTime()` promise with the last Promise.all in this function so it moreaccurately reflects the last update time.

Comment (chrismaree) [reply financial-templates-lib/clients/ExpiringMultiPartyEventClient.js]: ye, good point. I have implemented the change.

The monitor bot also implements this client and uses it to look for liquidations, disputes, dispute settlements & new sponsor events.

I have not validated that the daily reporter still runs but all the unit tests still work fine. I can run it to double-check but feel pretty confident seeing the tests all pass. What do you think?

Comment (nicholaspai) [reply financial-templates-lib/clients/ExpiringMultiPartyEventClient.js]: LGTM but yeah if possible can you run reporter locally as sanity check?

Comment (chrismaree) [reply financial-templates-lib/clients/ExpiringMultiPartyEventClient.js]: Tested and still workes exactly as expected :)

**Pull request merged**